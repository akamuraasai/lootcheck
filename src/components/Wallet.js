import React, { Component } from 'react';
import { connect } from 'react-redux';

import { depositar, sacar } from '../actions/balance';

export class Wallet extends Component {
  constructor() {
    super();

    this.state = { saldo: undefined };
  }

  atualizaSaldo = evt => {
    this.setState({ saldo: parseInt(evt.target.value, 10) });
  }

  depositar = () => {
    this.props.depositar(this.state.saldo);
  }

  sacar = () => {
    this.props.sacar(this.state.saldo);
  }

  render() {
    return (
      <div>
        <h3 className="saldo">Saldo da Carteira: {this.props.saldo}</h3>
        <br />
        <input className='input-carteira' onChange={this.atualizaSaldo} />
        <button className='btn-deposito' onClick={this.depositar}>Depositar (R$)</button>
        <button className="btn-saque" onClick={this.sacar}>Sacar (R$)</button>
      </div>
    );
  }
}

export default connect(
  state => ({ saldo: state.saldo }),
  { depositar, sacar }
)(Wallet);
