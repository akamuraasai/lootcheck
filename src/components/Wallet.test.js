import React from 'react';
import { shallow } from 'enzyme';
import { Wallet } from './Wallet';

describe('Wallet', () => {
  const mockDepositar = jest.fn();
  const mockSacar = jest.fn();
  const props = { saldo: 20, depositar: mockDepositar, sacar: mockSacar };
  const carteira = shallow(<Wallet {...props} />);

  it('renderiza corretamente', () => {
    expect(carteira).toMatchSnapshot();
  });

  it('mostra o saldo presente nas props', () => {
    expect(carteira.find('.saldo').text()).toEqual('Saldo da Carteira: 20');
  });

  it('cria um input para sacar ou depositar dinheiro na conta', () => {
    expect(carteira.find('.input-carteira').exists()).toBe(true);
  });

  describe('quando o usuario escreve no input da carteira', () => {
    const saldo = '25';

    beforeEach(() => {
      carteira.find('.input-carteira').simulate('change', { target: { value: saldo } });
    });

    it('atualiza o valor local do saldo no "state" e converte para numero', () => {
      expect(carteira.state().saldo).toEqual(parseInt(saldo, 10));
    });

    describe('e o usuario deseja fazer um deposito', () => {
      beforeEach(() => {
        carteira.find('.btn-deposito').simulate('click');
      });

      it('dispara a função "depositar()" recebida via props com o valor do saldo local', () => {
        expect(mockDepositar).toHaveBeenCalledWith(parseInt(saldo, 10));
      });
    });

    describe('e o usuario deseja fazer um saque', () => {
      beforeEach(() => {
        carteira.find('.btn-saque').simulate('click');
      });

      it('dispara a função "sacar()" recebida via props com o valor do saldo local', () => {
        expect(mockSacar).toHaveBeenCalledWith(parseInt(saldo, 10));
      });
    })
  });
});
