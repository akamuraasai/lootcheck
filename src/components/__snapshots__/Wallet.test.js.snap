// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renderiza corretamente 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    depositar={[Function]}
    sacar={[Function]}
    saldo={20}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="saldo"
>
          Saldo da Carteira: 
          20
</h3>,
        <br />,
        <input
          className="input-carteira"
          onChange={[Function]}
/>,
        <button
          className="btn-deposito"
          onClick={[Function]}
>
          Depositar
</button>,
        <button
          className="btn-saque"
          onClick={[Function]}
>
          Sacar
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Saldo da Carteira: ",
            20,
          ],
          "className": "saldo",
        },
        "ref": null,
        "rendered": Array [
          "Saldo da Carteira: ",
          20,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-carteira",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Depositar",
          "className": "btn-deposito",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Depositar",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sacar",
          "className": "btn-saque",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Sacar",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="saldo"
>
            Saldo da Carteira: 
            20
</h3>,
          <br />,
          <input
            className="input-carteira"
            onChange={[Function]}
/>,
          <button
            className="btn-deposito"
            onClick={[Function]}
>
            Depositar
</button>,
          <button
            className="btn-saque"
            onClick={[Function]}
>
            Sacar
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Saldo da Carteira: ",
              20,
            ],
            "className": "saldo",
          },
          "ref": null,
          "rendered": Array [
            "Saldo da Carteira: ",
            20,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-carteira",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Depositar",
            "className": "btn-deposito",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Depositar",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sacar",
            "className": "btn-saque",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Sacar",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
